From foo@baz Tue Oct  4 03:42:54 PM CEST 2022
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Tue, 4 Oct 2022 15:17:46 +0200
Subject: Revert "drm/i915/bios: Split parse_driver_features() into two parts"

From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

This reverts commit 5da3f1bfb88ed59e43d9b52529321c20199e8826 which is
commit c3fbcf60bc74b630967f291f47f0d9d0de6fcea7 upstream.

This is part of a series of i915 patches that were backported to 5.19.12
but found to be incomplete and caused problems on many systems so they
are being reverted.

Reported-by: Jerry Ling <jiling@cern.ch>
Reported-by: Hans de Goede <hdegoede@redhat.com>
Cc: Thorsten Leemhuis <regressions@leemhuis.info>
Cc: Ville Syrjälä <ville.syrjala@linux.intel.com>
Cc: Jani Nikula <jani.nikula@intel.com>
Cc: Sasha Levin <sashal@kernel.org>
Link: https://lore.kernel.org/all/55905860-adf9-312c-69cc-491ac8ce1a8b@cern.ch/
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/gpu/drm/i915/display/intel_bios.c |   11 -----------
 1 file changed, 11 deletions(-)

--- a/drivers/gpu/drm/i915/display/intel_bios.c
+++ b/drivers/gpu/drm/i915/display/intel_bios.c
@@ -1188,16 +1188,6 @@ parse_driver_features(struct drm_i915_pr
 		    driver->lvds_config != BDB_DRIVER_FEATURE_INT_SDVO_LVDS)
 			i915->vbt.int_lvds_support = 0;
 	}
-}
-
-static void
-parse_panel_driver_features(struct drm_i915_private *i915)
-{
-	const struct bdb_driver_features *driver;
-
-	driver = find_section(i915, BDB_DRIVER_FEATURES);
-	if (!driver)
-		return;
 
 	if (i915->vbt.version < 228) {
 		drm_dbg_kms(&i915->drm, "DRRS State Enabled:%d\n",
@@ -2975,7 +2965,6 @@ void intel_bios_init(struct drm_i915_pri
 	parse_lfp_backlight(i915);
 	parse_sdvo_panel_data(i915);
 	parse_driver_features(i915);
-	parse_panel_driver_features(i915);
 	parse_power_conservation_features(i915);
 	parse_edp(i915);
 	parse_psr(i915);
